hooks/node_handlers/base_export_handler.py

```diff
@@ -0,0 +1,644 @@
+        to inform the user.
+
+        :param parm: A :class:`hou.Parm` instance.
+
+        :returns: :class:`bool`. The status of the validation. True
+            is a pass.
+        """
+        value = parm.evalAsString().strip()
+        try:
+            self._validate_input(value)
+            return True
+        except FieldInputError as error:
+            parm.set("")
+            QtGui.QMessageBox.warning(
+                self.parent._get_dialog_parent(),
+                "Input Error",
+                str(error)
+            )
+            return False
+
+    def validate_parm_and_refresh_path(self, kwargs):
+        """
+        Callback to validate the parm and refresh the file path
+        accordingly.
+        """
+        if self._validate_parm(kwargs["parm"]):
+            self._refresh_file_path(kwargs["node"])
+
+    #############################################################################################
+    # Utilities
+    #############################################################################################
+
+    @staticmethod
+    def get_optional_keys(template):
+        """
+        Get the optional keys from the given template.
+
+        :param template: An :class:`sgtk.Template` instance.
+
+        :rtype: list(:class:`sgtk.TemplateKey`)
+        """
+        return filter(
+            template.is_optional,
+            template.keys.keys()
+        )
+
+    def _get_template_for_file_path(self, node, file_path):
+        """
+        Get the template for the given file path.
+        For the most part, this will liekly be the work_template.
+
+        :param node: A :class:`hou.Node` instance.
+        :param str file_path: The file path the check against.
+
+        :rtype: :class:`sgtk.Template`
+        """
+        return self.get_work_template(node)
+    
+    def _populate_from_fields(self, node, fields):
+        """
+        Populate the node from template fields.
+
+        :param node: A :class:`hou.Node` instance.
+        :param dict fields: The template fields.
+        """
+        sgtk_element = node.parm(self.SGTK_ELEMENT)
+        sgtk_element.set(fields.get("name", ""))
+
+        sgtk_location = node.parm(self.SGTK_LOCATION)
+        sgtk_location.set(fields.get("location", ""))
+
+        sgtk_variation = node.parm(self.SGTK_VARIATION)
+        sgtk_variation.set(fields.get("variation", ""))
+
+    def _set_version(self, node, current_version):
+        """
+        Set the index of the versions drop down from the given version.
+
+        :param node: A :class:`hou.Node` instance.
+        :param int current_version: The version to set.
+        """
+        version_str = str(current_version)
+        sgtk_version = node.parm(self.SGTK_VERSION)
+        entries = sgtk_version.menuItems()
+        if version_str not in entries:
+            index = len(entries)
+        else:
+            index = entries.index(version_str)
+        sgtk_version.set(index)
```

Comment:
switch it around to remove extra `not`

```suggestion
        if version_str in entries:
            index = entries.index(version_str)
        else:
            index = len(entries)
        sgtk_version.set(index)
```

Otherwise, you could also try just having it in 1 statement like:

```python
        sgtk_version.set(
            entries.index(version_str)
            if version_str in entries else
            len(entries)
        )
```